librz/core/cbin.c

```diff
@@ -2791,12 +2791,14 @@ static int bin_sections(RzCore *r, PJ *pj, int mode, ut64 laddr, int va, ut64 at
 		rz_flag_space_set(r->flags, print_segments ? RZ_FLAGS_FS_SEGMENTS : RZ_FLAGS_FS_SECTIONS);
 	}
 	if (IS_MODE_NORMAL(mode)) {
+		rz_cons_printf("Key to Flags:\n  W (write), A (alloc), X (execute), M (merge), S (strings), I (info),\n\
```

Comment:
this is probably just very specific to ELF, so it becomes useless for PE/Mach-O or any other file format.

Reply:
I propose to specify the legend in the `iS` command help. We can be more verbose there.

Reply:
Would you suggest that I return a RzList of flags since there are multiple flags possible for a single section and this is almost similar to the way permissions of the section are set?
I will have to add the flags for [PE/Macho], I didn't look at it initially since this issue was about adding info about ELF sections headers.
If I had to return int I would have to assign for all combinations of flags and map the int to a string again
```
# define WA 1
int get_section(RzBin *obj){
    if(SH_WRITE & SH_ALLOC){
        return WA;
    }
}
```
Then in the print function again :
```
#define WA "WA" or "write, alloc"

if(getsection(section->flag) ==1){
    return WA;
}

```

Reply:
> Would you suggest that I return a RzList of flags since there are multiple flags possible for a single section and this is almost similar to the way permissions of the section are set?

Not a RzList. It should be similar to how RzBinSection->flags works, that is that you just *or* (`|`) the flags. This approach has the limitation of having at most 64 flags (by using `ut64` to store the flags). If this is not enough, we could use a struct containing multiple bool with `:1` to make them use just 1 bit. @XVilka cc